modules/proto_tls/proto_tls.c

```diff
@@ -1343,7 +256,7 @@ static int tls_conn_init(struct tcp_connection* c)
 		if (!avp) {
 			LM_DBG("no TLS client doman AVP set, looking "
 				"for socket based TLS client domain\n");
-			dom = tls_find_client_domain(&c->rcv.src_ip, c->rcv.src_port);
+			dom = tls_mgm_api.find_client_domain(&c->rcv.src_ip, c->rcv.src_port);
 			if (dom) {
 				LM_DBG("found socket based TLS client domain "
 					"[%s:%d]\n", ip_addr2a(&dom->addr), dom->port);
```

Comment:
The tls_client_domain_avp should not be visible in the proto_tls module. The avp search should be done internally in the tls_mgm module and the "domain" should be returned directly.